util/info.c

```diff
@@ -233,24 +234,39 @@ static int print_vars(void)
 	int ret, count, i;
 	struct fi_param *params;
 	char delim;
+	char *var_prefix;
 
 	ret = fi_getparams(&params, &count);
 	if (ret)
 		return ret;
 
+	if (hints->fabric_attr->prov_name) {
+		if (asprintf(&var_prefix, "FI_%s", hints->fabric_attr->prov_name) < 0) {
+			var_prefix = NULL;
+			fi_freeparams(params);
+			return EXIT_FAILURE;
+		}
+		for (i = 0; i < var_prefix[i]; ++i)
+			var_prefix[i] = toupper((unsigned char) var_prefix[i]);
+	}
```

Comment:
This functionality would be better pulled out into it's own function or two.  One function would extract the provider name from the environment variable.